File: scripts/estacionariedad_estacionalidad.R
library(tseries)
library(forecast)
library(zoo)

setwd("C:/Users/hp master/OneDrive/Escritorio/series tiempo")

wn_rw <- read.csv("wn_rw.csv")
tabla_df <- read.csv("Tabla_rp.csv")
serie_covid <- read.csv("muertes_covid.csv")

tabla_df <- tabla_df[-c(1:24),]

tiempo_wn_rw <- seq(as.Date("2021/1/1"),as.Date("2021/12/31"),"days")
tiempo_tabla_df <- seq(as.Date("2012/1/31"),as.Date("2019/12/31"),"months")

tiempo_tabla_df


ruido_blanco <- zoo(x=wn_rw$ruido,order.by = tiempo_wn_rw)
caminata_aleatoria <- zoo(x=wn_rw$caminata,order.by = tiempo_wn_rw)

total_delitos <- zoo(x=tabla_df$totales,order.by = tiempo_tabla_df,frequency=12)

muertes_covid <- zoo(serie_covid$Total,as.Date(serie_covid$Fecha))

plot(ruido_blanco)
plot(caminata_aleatoria)

plot(total_delitos)

plot(muertes_covid)

########################

adf.test(ruido_blanco)
adf.test(caminata_aleatoria)

adf.test(total_delitos, k = ceiling(12*(length(total_delitos)/100)^(1/4)))

adf.test(muertes_covid)

########################

File: scripts/estacionariedad_estacionalidad.py
# -*- coding: utf-8 -*-
"""
Created on Fri Mar 26 15:32:54 2021

@author: hp
"""
#pip install statsmodels

import pandas as pd
import numpy as np
from datetime import datetime 
import os
import statsmodels.tsa.stattools as sts
from statsmodels.tsa.seasonal import seasonal_decompose

os.chdir("C:\\Users\\hp master\\OneDrive\\Escritorio\\series tiempo")

#%%

# Recordemos que Python empieza las numeraciones desde 0

delitos = pd.read_csv("Tabla_rp.csv")[24:]
delitos.totales.plot(figsize=(20,5),title="Total de delitos mes a mes por 8 años")

delitos


delitos.head()

delitos["fecha"]  = pd.date_range(pd.datetime(2012,1,1), periods=96, freq="M")

delitos.head()

delitos.fecha = pd.to_datetime(delitos.fecha, dayfirst = False)
delitos.set_index("fecha",inplace=True)

delitos.head()

delitos.totales
delitos.columns
#%%
delitos.totales.plot(figsize=(20,5),title="Robo parcial de vehículo")

sts.adfuller(delitos.totales,regression="ct"maxlag=???,autolag=None)

#%%

wn_rw = pd.read_csv("wn_rw.csv")
wn_rw["fecha"] = pd.date_range(pd.datetime(2021,1,1),periods=365,freq="d")


wn_rw.set_index("fecha",inplace=True)
sts.adfuller(delitos["totales"],regression="ct",maxlag=12,autolag=None)

delitos

#%%

'''Estacionalidad'''

descomposicion_aditiva_delitos_totales = seasonal_decompose(delitos["totales"],model="additive")
descomposicion_aditiva_delitos_totales.plot()

descomposicion_aditiva_ruido = seasonal_decompose(wn_rw["ruido"],model="additive")
descomposicion_aditiva_ruido.plot()

descomposicion_aditiva_caminata = seasonal_decompose(wn_rw["caminata"],model="additive")
descomposicion_aditiva_caminata.plot()



